### DISCUSSION TITLE
Require your help

### DISCUSSION THREAD

**Discussion by limkaflyer** [kubilus1/autoortho#13] [Q&A]
I have everything configured but X-Plane 12 is not downloading orthos. I am attaching a screenshot of the command prompt. Kindly assist please.
![Screenshot 2022-11-11 050328](https://user-images.githubusercontent.com/117994103/201347508-ac3179c4-2037-41a3-aab6-9415092a41f4.png)

**Comment by ivotaHR** [comment]
Hi.
Did you make looking coverage .dsf files from Ortho4XP ?

**Reply by limkaflyer** [reply]
Yes I have all the necessary files from Ortho4xp which includes dsf files.

**Comment by kubilus1** [comment]
The warning indicates the orthos are your issue.   Take a look at some of
my sample provided Ortho packages.   You can't just directly take ortho4xp
output, you must combine the various dirs.

**Reply by limkaflyer** [reply]
I am getting same warning with your provided Utah package.

**Comment by kubilus1** [comment]
Check what the directory that is created in customer scenery looks like.
 That warning is just a simple check of directory structure.

**Reply by limkaflyer** [reply]
This is what it looks like
![2](https://user-images.githubusercontent.com/117994103/201402571-a1290eb8-a090-47a2-b5a1-40f9556c5805.png)

**Comment by kubilus1** [comment]
Interesting, that looks correct.  I wonder if Xplane is doing
something different detecting directories in X-Plane 12 on Windows
specifically.

**Reply by limkaflyer** [reply]
Yeah I am pretty sure I did everything correct. I went through the entire process multiple times. The programs kind of works but its not fetching orthos. My network activity also remained at zero. Anyways please do let me know if you find a workaround/fix. Would love to revisit this.
Thanks

**Comment by kubilus1** [comment]
Actually, can you show what is under the 'Earth nav data' folder from the
source of our orthos dir?  Basically you should have a structure like:
```
Earth nav data/
├── +30-110
│   ├── +37-110.dsf
│   ├── +38-110.dsf
│   └── +39-110.dsf
├── +30-120
│   ├── +37-111.dsf
│   ├── +37-112.dsf
│   ├── +37-113.dsf
│   ├── +37-114.dsf
│   ├── +38-111.dsf
│   ├── +38-112.dsf
│   ├── +38-113.dsf
│   ├── +38-114.dsf
│   ├── +39-111.dsf
│   ├── +39-112.dsf
│   ├── +39-113.dsf
│   └── +39-114.dsf
├── +40-110
│   └── +40-110.dsf
└── +40-120
    ├── +40-111.dsf
    ├── +40-112.dsf
    ├── +40-113.dsf
    ├── +40-114.dsf
    ├── +41-112.dsf
    ├── +41-113.dsf
    └── +41-114.dsf
```

You should see that in both your orthos dir, and also in the mount under
Custom Scenery.  The fact the autoortho is complaining about the structure,
and XPlane isn't finding the DSF files, tells me this is probably not quite
right.

**Reply by limkaflyer** [reply]
I have that exact same structure in my ortho folder. But, the folder that is inside the custom scenery folder, when I go inside that folder and click on the shortcut of earth nav data, it gives me this error 
![Screenshot 2022-11-12 110350](https://user-images.githubusercontent.com/117994103/201490594-84147def-826d-48f0-9379-37c50c9f0984.png)

**Reply by kubilus1** [reply]
So is the source dir `E:/z_autoortho` a NAS mount or perhaps a shortcut instead of the full path?  That 'Earth nav data' link is simply just a Windows junction, no magic there.  I could see Windows complaining if this is not a full path or it's not a local drive.

**Reply by limkaflyer** [reply]
Not a NAS mount and not a shortcut either. From that error 'location is not available' it seems that the E:/z_autoortho folder is not accessible which explains further why X-Plane is not able to access the files and hence no ortho streaming. This should not happen as I have all my orthos stored in the 'E" drive while the Xplane itself is in 'D' but Xplane is able access all those files normally (have the symlink and shortcuts of ortho folders in D:Xplane/custom scenery). So I am very sure that disk drives are ok. 

One thing I have noticed when I click on the properties of the 'Earth nav data' and 'terrain' shortcut folders I see that they donot have a target folder ? This is not correct I assume ? It should have a target folder.
![Screenshot 2022-11-12 110350](https://user-images.githubusercontent.com/117994103/201523633-c4ffff7b-c71b-49f6-8df4-d1a0ae7cedd5.png)

**Reply by limkaflyer** [reply]
This is what a normal ortho shortcut looks like. See it has the target folder.
![Screenshot 2022-11-13 0517083](https://user-images.githubusercontent.com/117994103/201523720-f2712e18-d4bb-46bf-acd2-dfab2ba770f4.png)

**Reply by kubilus1** [reply]
This uses a filesystem junction, not a shortcut, so that may be why the properties look different.  Nothing too special here, this is where the directories are configured:  https://github.com/kubilus1/autoortho/blob/main/autoortho/aoconfig.py#L224

All this does is check for the existence of the `z_autoorotho/Earth nav data` and `z_autoortho/terrain` dirs, and just simply does a `mklink /J "dest" "src"`.  So it's really an operating system issue.

What I would try is just deleting the z_autoortho directory in Custom Scenery.  Autoortho will recreate this next run.  Perhaps you moved the source directory of orthos around at some point, so now this junction doesn't point anywhere.  That appears to be what windows is telling you in one of the earlier screenshots.  Perhaps there are some permissions preventing things from working on your computer?

**Comment by kubilus1** [comment]
Yeah eventually I want this to just set all the orthos up for you
automagically.  But I'll have to have the time to do that.

**Comment by ivotaHR** [comment]
@limkaflyer  it seems you had wrong ortho files. You used folders from XP11 scenery.  You have multiple folders zOrtho4XP_XXXXXX.
You should make only one folder.

You should make folder ORTHO_TITLES. the put folders like this:
Earth nav data/
├── +30-110
│   ├── +37-110.dsf
│   ├── +38-110.dsf
│   └── +39-110.dsf
├── +30-120
│   ├── +37-111.dsf
│   ├── +37-112.dsf
│   ├── +37-113.dsf
│   ├── +37-114.dsf
│   ├── +38-111.dsf
│   ├── +38-112.dsf
│   ├── +38-113.dsf
│   ├── +38-114.dsf
│   ├── +39-111.dsf
│   ├── +39-112.dsf
│   ├── +39-113.dsf
│   └── +39-114.dsf
├── +40-110
│   └── +40-110.dsf
└── +40-120
    ├── +40-111.dsf
    ├── +40-112.dsf
    ├── +40-113.dsf
    ├── +40-114.dsf
    ├── +41-112.dsf
    ├── +41-113.dsf
    └── +41-114.dsf
```

**Reply by kubilus1** [reply]
I think that screen shot with the multiple zOrthoXXXXX files was just the overall Custom Scenery location.  Presumably with other manually generated orthos.  So that is all fine, I think.
